Sources/Util/Formatting.swift

```diff
@@ -62,8 +62,34 @@ extension TracedProcessor {
        return ""
      }
      return "\(registers)\n"
-   }
-  public func formatInput() -> String{
+  }
+
+  public func formatInput() -> String {
+    // String override for printing sub-character information.
+    if !input.indices.contains(currentPosition),
+       let input = input as? StringProtocol {
```

Comment:
In swift 5.5, I get an error on this line:

> error: protocol 'StringProtocol' can only be used as a generic constraint because it has Self or associated type requirements

Reply:
This fixes it (warning, commit is from a very broken branch, but should be cherry-pickable for anyone blocked on this if I'm not able to land it myself soon):

https://github.com/milseman/swift-experimental-string-processing/commit/15b3112c8217943db30e6232cfa0c30ab7b6e3fc

Reply:
I think I'm using a newer compiler. Strange that CI didn't catch it.

Reply:
https://github.com/apple/swift-experimental-string-processing/pull/51